packages/utils/src/build-operation-for-field.ts

```diff
@@ -284,6 +284,12 @@ function resolveSelectionSet({
   }
 
   if (isObjectType(type)) {
+    // Do not resolve `Query` type
+    // More info: https://github.com/ardatan/graphql-tools/issues/3380
+    if (type.name === 'Query') {
```

Comment:
Instead of checking if name is "Query", maybe we should check if it is a root type.
What do you think?

Reply:
I did not find `isQueryType` or `isRootType`. Are you know what can I use? I can do something like this `if (type === schema.getQueryType()){`. Any ideas?